tests/unit/test_space.py

```diff
@@ -540,6 +591,56 @@ def test_product_space_get_subspace() -> None:
     npt.assert_array_equal(subspace_C.upper, [2])
 
 
+@pytest.mark.parametrize(
+    "points",
+    [
+        tf.ones((1, 5), dtype=tf.float64),
+        tf.ones((2, 3), dtype=tf.float64),
+    ],
+)
+def test_product_space_fix_subspace_fixes_desired_subspace(points: tf.Tensor) -> None:
+    spaces = [
+        Box([-1, -2], [2, 3]),
+        DiscreteSearchSpace(tf.constant([[-0.5]], dtype=tf.float64)),
+        Box([-1], [2]),
+    ]
+    tags = ["A", "B", "C"]
+    product_space = TaggedProductSearchSpace(spaces=spaces, tags=tags)
+
+    for tag in tags:
+        product_space_with_fixed_subspace = product_space.fix_subspace(tag, points)
+        new_subspace = product_space_with_fixed_subspace.get_subspace(tag)
+        assert isinstance(new_subspace, DiscreteSearchSpace)
+        npt.assert_array_equal(new_subspace.points, points)
+
+
+@pytest.mark.parametrize(
+    "points",
+    [
+        tf.ones((1, 5), dtype=tf.float64),
+        tf.ones((2, 3), dtype=tf.float64),
+    ],
+)
+def test_product_space_fix_subspace_doesnt_fix_undesired_subspace(points: tf.Tensor) -> None:
+    spaces = [
+        Box([-1, -2], [2, 3]),
+        DiscreteSearchSpace(tf.constant([[-0.5]], dtype=tf.float64)),
+        Box([-1], [2]),
+    ]
+    tags = ["A", "B", "C"]
+    product_space = TaggedProductSearchSpace(spaces=spaces, tags=tags)
+
+    for tag in tags:
+        product_space_with_fixed_subspace = product_space.fix_subspace(tag, points)
+        remaining_tags = tags
+        remaining_tags.remove(tag)
+        for non_target_tag in remaining_tags:
```

Comment:
Simpler to just use an if test:

```suggestion
        for other_tag in tags:
            if other_tag != tag:
```